static/app/utils/replays/getCurrentUserAction.tsx

```diff
@@ -4,9 +4,9 @@ import {Crumb} from 'sentry/types/breadcrumbs';
 export function getCurrentUserAction(
   userActionCrumbs: Crumb[] | undefined,
```

Comment:
this is probably correct as-is, there's a slight difference between `?` and explicit `undefined`, where the former means the argument is optional and the latter means you should be expected to pass an argument, but its value can be undefined.

Reply:
so, in terms of clarity and fixes, should I leave as it's right now or should I change @billyvg ?

Reply:
`Crumb[] | undefined` is the correct typing i think. @billyvg is right about the subtle difference with `?` and we do want the prop to be explicitly listed, even if the value is not a `Crumb[]` or `number` or whatever the case may be.

All three of these props should use `| undefined` and should not use `?`.

---

As far as `undefined` vs. `null`

JS sucks and why are there two ways to spell 'no value' it is just the worst to deal with.
BUT! In my experience TS makes it easy to choose `undefined` for cases like this  because methods in the standard lib and new syntax return `undefined` in most cases.

What matters is what does this this case specifically do? For these values the optional-chaining `?.` operator is how we get the values originally. Something like `<Foo crumbs={replay?.getRawCrumbs()} />`.
This means the type up the component tree is `Crumbs[] | undefined` so it's much easier to pass that along into all the downstream components. We should prefer `| undefined` and not `| null` because it's error prone to convert from undefined to null all the time.